### ISSUE TITLE
Linux maintaining compliance

### ISSUE THREAD

**Issue by tieugene** [pawelsalawa/sqlitestudio#3871]
During packaging subj into rpm I found next ~bugs~ chellenges:
- *.qm in source (they must be produced with 'lrelease *.pro' during build stage)
- translations not installing
- Plugins' 'make install' not works
- freedesktop.org entries no present:
  - *.desktop
  - icons
  - *.appdata.xml
- <libdir>/*.so - ordinary they must be in *-devel package, but there is no sence without *.h, that are not installing.
- libriries installing exactly into /usr/lib; modern x64 linuxes use /usr/lib64
- cannot start:
```
bash-5.0$ sqlitestudio
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.
```
But it is compilable, thank you.

**Comment by pawelsalawa** [comment]
Qt Xcb plugin failed to load. Do you have several Qt versions installed? It looks like your Qt has some issues.

**Comment by tieugene** [comment]
> Qt Xcb plugin failed to load. Do you have several Qt versions installed? It looks like your Qt has some issues.

No, the only qt5-5.15.1 from official Fedora 33 repo (in rpms).
And no one application from sources. RPMs only.

**Comment by pawelsalawa** [comment]
Please have a look if this helps: https://forum.qt.io/topic/93247/qt-qpa-plugin-could-not-load-the-qt-platform-plugin-xcb-in-even-though-it-was-found/5

**Comment by tieugene** [comment]
> Please have a look if this helps: https://forum.qt.io/topic/93247/qt-qpa-plugin-could-not-load-the-qt-platform-plugin-xcb-in-even-though-it-was-found/5

I cannot repeat the bug - application starts ok.
(?..)

Another problem - now I cannot close it in normal way.
There is no "File - Quit" or something like this.
Is it bug or feature?

**Comment by srgonzalezz** [comment]
In my case, I use a debian distribution. I installed [Libxcb](http://xcb.freedesktop.org/) following the instructions [here](https://wiki.qt.io/Building_Qt_5_from_Git#Linux/X11) under the header of Libxcb.

Ubuntu/Debian:
`sudo apt-get install '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev`

Other distributions are listed there.

**Comment by Ashvith10** [comment]
I was able to make it work with Wayland on Guix without having to require `libxcb`. This particular line is what's causing issue on most distros, you may want to patch it:

https://github.com/pawelsalawa/letos/blob/ac2020e7f1223d52eb17f3670d9b750113c6ef3a/Letos/letos/main.cpp#L137

Additionally, depending on the build project, you will also need to set the config flag `-DSYS_PLUGINS_DIR` (this is essential during the build phase, as it will hardcode the system plugin path) and `LETOS_PLUGINS`, a temporary environment variable during the test phase to point to the plugins directory.

In my case, since I had the Letos plugins exist as a separate package, and built BEFORE Letos itself, both `-DSYS_PLUGINS_DIR` and `LETOS_PLUGINS` points to the same directory (`#~(string-append #$(this-package-input "letos-plugins") "/lib/letos"))`).

**Comment by pawelsalawa** [comment]
The
```
qunsetenv("QT_PLUGIN_PATH");
```
is necessary and even is commented to remind why it is necessary.
